src/msw/overlay.cpp

```diff
@@ -0,0 +1,228 @@
+            }
+
+            gs_overlayClassName = nullptr;
+        }
+
+        return true;
+    }
+
+    virtual bool Show(bool show) override
+    {
+        if ( !wxWindowBase::Show(show) || !GetHandle() )
+            return false;
+
+        if ( show )
+        {
+            ::SetWindowPos(GetHandle(), nullptr, 0, 0, 0, 0,
+                           SWP_NOSIZE |
+                           SWP_NOMOVE |
+                           SWP_NOREDRAW |
+                           SWP_NOOWNERZORDER |
+                           SWP_NOACTIVATE |
+                           SWP_SHOWWINDOW);
+        }
+        else
+        {
+            ::ShowWindow(GetHandle(), SW_HIDE);
+        }
+
+        return true;
+    }
+
+private:
+    wxOverlayWindow(HWND hwnd) : wxNativeContainerWindow(hwnd) {}
+    ~wxOverlayWindow() {}
+};
+} // anonymous namespace
+
+class wxOverlayImpl : public wxOverlay::Impl
+{
+public:
+    wxOverlayImpl();
+    ~wxOverlayImpl();
+
+    virtual bool IsNative() const override { return true; }
+    virtual void Reset() override;
+    virtual bool IsOk() override;
+    virtual void Init(wxDC* dc, int x , int y , int width , int height) override;
+    virtual void BeginDrawing(wxDC* dc) override;
+    virtual void EndDrawing(wxDC* dc) override;
+    virtual void Clear(wxDC* dc) override;
+
+public:
+    // window the overlay is associated with
+    wxWindow* m_window = nullptr;
+    // the overlay window itself
+    wxWindow* m_overlayWindow = nullptr;
+
+    wxRect m_rect;
+
+    wxDECLARE_NO_COPY_CLASS(wxOverlayImpl);
+};
+
+wxOverlayImpl::wxOverlayImpl()
+{
+}
+
+wxOverlayImpl::~wxOverlayImpl()
+{
+    Reset();
+}
+
+bool wxOverlayImpl::IsOk()
+{
+    return m_overlayWindow != nullptr;
+}
+
+void wxOverlayImpl::Init(wxDC* dc, int , int , int , int )
+{
+    if ( IsOk() )
+        return;
+
+    m_window = dc->GetWindow();
+    m_overlayWindow = wxOverlayWindow::Create();
+
+    m_rect.SetSize(m_window->GetClientSize());
+    m_rect.SetPosition(m_window->GetScreenPosition());
+
+    m_overlayWindow->Move(m_rect.GetPosition());
+    m_overlayWindow->SetSize(m_rect.GetSize());
+    m_overlayWindow->Show();
+}
+
+void wxOverlayImpl::BeginDrawing(wxDC* WXUNUSED(dc))
+{
+    // nothing to do here
+}
+
+void wxOverlayImpl::EndDrawing(wxDC* dc)
+{
+    wxCHECK_RET( IsOk(), wxS("overlay not initialized") );
+
+    wxWindowDC winDC(m_overlayWindow);
+
+    const wxPoint pt = dc->GetDeviceOrigin();
+    winDC.Blit(0, 0, m_rect.width, m_rect.height, dc, -pt.x, -pt.y);
+}
+
+void wxOverlayImpl::Clear(wxDC* dc)
+{
+    wxCHECK_RET( IsOk(), wxS("overlay not initialized") );
+
+    dc->SetBackground(wxBrush(wxTransparentColour));
+    dc->Clear();
+}
+
+void wxOverlayImpl::Reset()
+{
+    if ( IsOk() )
+    {
+        m_overlayWindow->Destroy();
+        m_overlayWindow = nullptr;
+        m_window = nullptr;
+    }
+}
+
+wxOverlay::Impl* wxOverlay::Create()
+{
+    wxString value;
```

Comment:
```suggestion
```
Sorry for all the nitpicking. 

Talking about nitpicking, I would also unify how Win32 function names are referred in the `wxLogLastError()` messages: Unlike in `wxOverlayWindow::Destroy()`, in `wxOverlayWindow::Create()`  the parentheses after the function name are missing.

Reply:
> feel free to leave any comments even if it's just a nitpick.

Be careful what you wish for. ;) OK, the last one, I promise. The file header in `src/msw/overlay.cpp` misses _Purpose_ and _Created_ lines (and Author's e-mail). 

Not sure if all these fields are still useful, but that's what the current coding guidelines [say](https://wxwidgets.org/develop/coding-guidelines/#wxwin_header).